enumerable_methods.rb

```diff
@@ -0,0 +1,212 @@
+module Enumerable
+  # my_each
+
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    arr = *self
+    i = 0
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    arr = *self
+    i = 0
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+
+  # select
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    arr = *self
+    new_arr = []
+    arr.my_each do |element|
+      next unless yield(element)
+
+      new_arr.push(element)
+    end
+    new_arr
+  end
+
+  # my_all?
+
+  def my_all?(arg = nil)
+    condition = true
+    my_each do |element|
+      if arg.is_a?(Integer)
+        if element != arg
+          condition = false
+          break
+        end
+      end
+      if arg.is_a?(Class)
+        if element.is_a?(arg) == false
+          condition = false
+          break
+        end
+      elsif arg.is_a?(Regexp)
+        if element.match?(arg) == false
+          condition = false
+          break
+        end
+      else
+        if block_given?
+          if yield(element) == false || element.nil?
+            condition = false
+            break
+          end
+        else
+          if element == false || element.nil?
+            condition = false
+            break
+          end
+        end
+      end
+    end
+    condition
+  end
+
+  # my_any?
+
+  def my_any?(arg = nil)
```

Comment:
- When no block or argument is given, **returns true** if at least one of the collections is not false or nil but your method returned `false`.

- When a pattern other than Regex or a Class is given, **returns true** if any of the collection matches the pattern but your method returned `false`.